promql/engine.go

```diff
@@ -1852,6 +1858,26 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, annotations.Annotatio
 	panic(fmt.Errorf("unhandled expression of type: %T", expr))
 }
 
+func reuseOrGetHPointSlices(prevSS *Series, numSteps int) (r []HPoint) {
```

Comment:
Suggest adding a descriptive comment, e.g.
// If the previous slice has lots of room, use that space for the new slice and adjust the capacity of the old slice so when it is re-used from the pool it doesn't overflow into the new one.

Reply:
PTAL?